src/improvements/template/SetDisputeGameImpl.sol

```diff
@@ -0,0 +1,111 @@
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](1);
+        storageWrites[0] = "DisputeGameFactoryProxy";
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with implementation configurations from a TOML file.
+    function _templateSetup(string memory taskConfigFilePath) internal override {
+        super._templateSetup(taskConfigFilePath);
+        string memory toml = vm.readFile(taskConfigFilePath);
+
+        GameImplConfig[] memory configs = abi.decode(toml.parseRaw(".gameImpls.configs"), (GameImplConfig[]));
+        for (uint256 i = 0; i < configs.length; i++) {
+            cfg[configs[i].chainId] = configs[i];
+        }
+    }
+
+    /// @notice Write the calls that you want to execute for the task.
+    function _build() internal override {
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        for (uint256 i = 0; i < chains.length; i++) {
+            uint256 chainId = chains[i].chainId;
+            GameImplConfig memory c = cfg[chainId];
+
+            address dgf = superchainAddrRegistry.getAddress("DisputeGameFactoryProxy", chainId);
+            DisputeGameFactory factory = DisputeGameFactory(dgf);
+
+            // Set FDG (CANNON) implementation if TOML is different from current on-chain
+            address currentFDG = address(factory.gameImpls(GameTypes.CANNON));
+            if (currentFDG != c.fdgImpl) {
+                factory.setImplementation(GameTypes.CANNON, IFaultDisputeGame(c.fdgImpl));
+            }
+
+            // Set PDG (PERMISSIONED_CANNON) implementation if TOML is different from current on-chain
+            address currentPDG = address(factory.gameImpls(GameTypes.PERMISSIONED_CANNON));
+            if (currentPDG != c.pdgImpl) {
+                factory.setImplementation(GameTypes.PERMISSIONED_CANNON, IPermissionedDisputeGame(c.pdgImpl));
+            }
+        }
+    }
+
+    /// @notice This method performs all validations and assertions that verify the calls executed as expected.
+    function _validate(VmSafe.AccountAccess[] memory, Action[] memory) internal view override {
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        for (uint256 i = 0; i < chains.length; i++) {
+            uint256 chainId = chains[i].chainId;
+            GameImplConfig memory c = cfg[chainId];
+
+            address dgf = superchainAddrRegistry.getAddress("DisputeGameFactoryProxy", chainId);
+            DisputeGameFactory factory = DisputeGameFactory(dgf);
+
+            // Assert FDG (CANNON) implementation matches TOML
+            assertEq(address(factory.gameImpls(GameTypes.CANNON)), c.fdgImpl);
+
+            // Assert PDG (PERMISSIONED_CANNON) implementation matches TOML
+            assertEq(address(factory.gameImpls(GameTypes.PERMISSIONED_CANNON)), c.pdgImpl);
```

Comment:
Same as the comment above, can we lean on the proofs team here to make sure we're performing enough checks on the resulting upgrade. I think being overly conservative here on the checks we want is a good first place to start.